### PR TITLE
Plugins marketplace: Change formatting of search and filtering

### PR DESCRIPTION
**Pull request by xavier-lc**
Related to https://github.com/Automattic/dotcom-forge/issues/7537

## Proposed Changes

* Update the design of the search and filters on `/plugins` to the one proposed on the linked issue.
* Figma: https://www.figma.com/design/kyCC1EHKtv4qsElmlAzp5T/Flyout-panel-i2?node-id=537-123318&m=dev

## Testing Instructions

* Open the live preview and go to `/plugins`
* Check that the search and filters still work, and look like the proposed design

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/plugins/plugins-browser/index.jsx [hunk 1]
```diff
@@ -171,33 +167,15 @@ const PluginsBrowser = ( { trackPageViews = true, category, search, hideHeader }
...
-							: __( 'Plugins you need to get your projects done' )
-					}
-					subtitle={
-						! isLoggedIn &&
-						( 'en' === locale ||
-							hasTranslation(
-								'Add new functionality and integrations to your site with thousands of plugins.'
-							) ) &&
-						__( 'Add new functionality and integrations to your site with thousands of plugins.' )
-					}
```
Comment (xavier-lc) [inline client/my-sites/plugins/plugins-browser/index.jsx:193]: Title and subtitle are no longer needed.

File: client/my-sites/plugins/search-categories/index.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+} from 'calypso/my-sites/plugins/categories/use-categories';
+import { useGetCategoryUrl } from 'calypso/my-sites/plugins/categories/use-get-category-url';
+import { useLocalizedPlugins } from 'calypso/my-sites/plugins/utils';
+import { recordTracksEvent, recordGoogleEvent } from 'calypso/state/analytics/actions';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { useTermsSuggestions } from './useTermsSuggestions';
+
+import './style.scss';
+
+const SearchBox: FC< {
```
Comment (xavier-lc) [inline client/my-sites/plugins/search-categories/index.tsx:25]: Component taken from the `search-box-header` folder.

File: client/my-sites/plugins/search-categories/index.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+			recordEvent={ recordSearchEvent }
+			searching={ isSearching }
+			submitOnOpenIconClick
+			openIconSide="right"
+			displayOpenAndCloseIcons
+		/>
+	);
+};
+
+const SearchCategories: FC< {
```
Comment (xavier-lc) [inline client/my-sites/plugins/search-categories/index.tsx:86]: The search logic has also been taken from the `search-box-header` folder.

**Comment by BogdanUngureanu** [comment]
The design looks great on desktop resolutions. However, on mobile, it looks a bit off to me. 
Screenshot 2024-06-07 at 11 42 41

Is that expected? I couldn't find a mockup for mobile resolutions in the Figma file. 

Maybe we want to increase the size of the inputs?

**Comment by mmtr** [comment]
This will also change the logged-out plugins marketplace:

Before | After
--- | ---
Screenshot 2024-06-07 at 14 09 36 | Screenshot 2024-06-07 at 14 09 39

Is that intentional? Looking at the issue, it seems we want to target the design to the logged-in page only.

**Comment by xavier-lc** [comment]
> Is that expected? I couldn't find a mockup for mobile resolutions in the Figma file.

There is none 😅

> Maybe we want to increase the size of the inputs?

Let me try 👍

**Comment by xavier-lc** [comment]
@BogdanUngureanu I've widened the input and adjusted the size of the button.

**Comment by BogdanUngureanu** [comment]
The changes look great on mobile now, thanks! The only thing that remains is the logged-out view and we can ship it!

**Comment by xavier-lc** [comment]
@BogdanUngureanu the logged out page is back to how it was.

**Comment by mmtr** [comment]
Found a few differences when comparing with production.

While logged-out, the top navigation bar was blue, it is now black:

Before | After
--- | ---
Screenshot 2024-06-11 at 13 09 05 | Screenshot 2024-06-11 at 13 09 13


While logged in, if you reload the page, there is a brief moment where the logged-out page shows up which does not happen in prod:

**Before**

https://github.com/Automattic/wp-calypso/assets/1233880/b0638e4b-80fc-48f1-b747-36d573bde221

**After**


https://github.com/Automattic/wp-calypso/assets/1233880/a5283718-363b-43a9-8662-8bee10097e50

**Comment by mmtr** [comment]
On mobile viewports, it's a bit weird to have a dropdown that says "More" with no additional context:

Screenshot 2024-06-11 at 13 15 33

It could maybe display the current selected option.

**Comment by mmtr** [comment]
This is also breaking a few E2E tests:

> specs/plugins/plugins__browse.ts: Plugins: Browse: Can browse SEO category
> specs/plugins/plugins__search.ts: Plugins Search: Navigate back to the default plugins page when searching from categories pages

File: client/my-sites/plugins/search-categories/index.tsx (hunk 2 above)
Comment (mmtr) [reply client/my-sites/plugins/search-categories/index.tsx:25]: What's preventing you using from the existing component?

File: client/my-sites/plugins/search-categories/index.tsx (hunk 3 above)
Comment (mmtr) [reply client/my-sites/plugins/search-categories/index.tsx:86]: Same as above, what's preventing you from using the existing logic?

**Comment by xavier-lc** [comment]
> While logged-out, the top navigation bar was blue, it is now black:

Fixed after rebase.

> While logged in, if you reload the page, there is a brief moment where the logged-out page shows up which does not happen in prod:

I think that's got to do with the dev environment. I see the same behavior on `trunk`, and I've also checked the live preview for 2 unrelated PRs and the same happens there.

**Comment by mmtr** [comment]
> I think that's got to do with the dev environment. I see the same behavior on trunk, and I've also checked the live preview for 2 unrelated PRs and the same happens there.

Thanks for confirming!

The only remaining blockers are the E2E failures.

The one about the dropdown on mobile can be addressed on a follow-up.

**Comment by xavier-lc** [comment]
> On mobile viewports, it's a bit weird to have a dropdown that says "More" with no additional context:

Yeah, it's how the `DropdownGroup` component [works](https://github.com/Automattic/wp-calypso/blob/trunk/packages/components/src/responsive-toolbar-group/dropdown-group.tsx#L108). I could leave it as it currently works with a swiper below the search field.

**Comment by xavier-lc** [comment]
> The only remaining blockers are the E2E failures.

Fixed now 🥳️

**Comment by taipeicoder** [comment]
Jotting down some minor UI issues in the logged-in `/plugins/:site`

Subtle border.
Screenshot 2024-06-12 at 2 35 29 PM

Outline color in selected category.
Screenshot 2024-06-12 at 12 18 19 PM

Odd categories that are not present in production?
Screenshot 2024-06-12 at 12 19 20 PM

**Comment by xavier-lc** [comment]
> Subtle border.

Good catch, fixed on https://github.com/Automattic/wp-calypso/pull/91565/commits/00040f6085cee8ac554ebe9259b43e62100ca91f

> Outline color in selected category.

Good catch, fixed on https://github.com/Automattic/wp-calypso/pull/91565/commits/5815e2c898d727011a2f02f50fe5db8cd814ea9c

> Odd categories that are not present in production?

The logic that handles the categories hasn't been changed. I'm not sure why that category is not on the dropdown, but it does show up on the main page:

image

You can see it on our codebase as well:

https://github.com/Automattic/wp-calypso/blob/f65ce82fa09fc78f424571c25f900b550d471445/client/my-sites/plugins/categories/use-categories.tsx#L332-L334

> Container width jumping around when switching from Discover to another category:

This has been discussed with @lucasmendes-design (see https://github.com/Automattic/dotcom-forge/issues/7491#issuecomment-2155360441) and will be addressed on a followup PR.

**Comment by mmtr** [comment]
> The logic that handles the categories hasn't been changed. I'm not sure why that category is not on the dropdown, but it does show up on the main page:

Presumably, we're hiding the categories that are displayed as collections in the discover page. We should keep that behavior

**Comment by mmtr** [comment]
> Presumably, we're hiding the categories that are displayed as collections in the discover page. We should keep that behavior

I guess the fact that the behavior is now accidentally different, is a side effect of duplicating some logic into the new `search-categories` folder. Can we re-use the existing logic instead of duplicating it?

**Comment by taipeicoder** [comment]
> Presumably, we're hiding the categories that are displayed as collections in the discover page. We should keep that behavior

+1. It's odd that we show them in the desktop viewport:

Screenshot 2024-06-12 at 9 43 03 PM

But in the mobile viewport we show the dropdown, which doesn't have those categories:

Screenshot 2024-06-12 at 9 42 57 PM

**Comment by xavier-lc** [comment]
Found it, I needed to use the `menu` key from the `Category` objects instead of the `title` one. Fixed on https://github.com/Automattic/wp-calypso/pull/91565/commits/d6177baac609a9c723190abd14b51c8056920e23

**Review by taipeicoder — APPROVED** [review]
Thanks for working on this, the changes LGTM 👍

**Pull request merged**